contracts/RollupPoB.sol

```diff
@@ -0,0 +1,331 @@
+        withdrawBid[msg.sender] = 0;
+        msg.sender.transfer(auxAmount);
+    }
+
+    /**
+     * @dev operator commits data and forge a batch
+     * @param compressedTx data committed by the operator. Represents off-chain transactions
+     * @param proofA zk-snark input
+     * @param proofB zk-snark input
+     * @param proofC zk-snark input
+     * @param input public zk-snark inputs
+     */
+    function commitAndForge(
+        bytes calldata compressedTx,
+        uint[2] calldata proofA,
+        uint[2][2] calldata proofB,
+        uint[2] calldata proofC,
+        uint[8] calldata input
+    ) external virtual {
+        uint32 slot = currentSlot();
+        Operator storage op = slotWinner[slot];
+        // message sender must be the controller address
+        require(msg.sender == op.forgerAddress, 'message sender must be forgerAddress');
+        // Check if deadline has been achieved to not commit any more data
+        uint blockDeadline = getBlockBySlot(slot + 1) - SLOT_DEADLINE;
+        require(getBlockNumber() < blockDeadline, 'not possible to commit data after deadline');
+        uint256 offChainHash = hashOffChainTx(compressedTx, MAX_TX);
+        emit dataCommitted(offChainHash);
+        // Check input off-chain hash matches hash commited
+        require(offChainHash == input[offChainHashInput],
+            'hash off chain input does not match hash commited');
+        rollupInterface.forgeBatch(op.beneficiaryAddress, proofA, proofB, proofC, input);
+        // one block has been forged in this slot
+        infoSlot[slot].fullFilled = true;
+    }
+
+     /**
+     * @dev commitAndForge after deadline
+     * @param compressedTx data committed by the operator. Represents off-chain transactions
+     * @param proofA zk-snark input
+     * @param proofB zk-snark input
+     * @param proofC zk-snark input
+     * @param input public zk-snark inputs
+     */
+    function commitAndForgeDeadline(
+        bytes calldata compressedTx,
+        uint[2] calldata proofA,
+        uint[2][2] calldata proofB,
+        uint[2] calldata proofC,
+        uint[8] calldata input
+    ) external virtual {
+        uint32 slot = currentSlot();
+        // Check if deadline has been achieved to not commit any more data
+        uint blockDeadline = getBlockBySlot(slot + 1) - SLOT_DEADLINE;
+        require(getBlockNumber() >= blockDeadline, 'not possible to commit data before deadline');
+        // Check there is no data to be forged
+        require(!infoSlot[slot].fullFilled, 'another operator has already forged data');
+        uint256 offChainHash = hashOffChainTx(compressedTx, MAX_TX);
+        emit dataCommitted(offChainHash);
+        // Check input off-chain hash matches hash commited
+        require(offChainHash == input[offChainHashInput],
+            'hash off chain input does not match hash commited');
+        rollupInterface.forgeBatch(msg.sender, proofA, proofB, proofC, input);
```

Comment:
- Missing one parameter at the end of `forgeBatch` function:
  - `compressedOnChainTx`

Applies to `commitAndForge` function as well